Composer/packages/server/src/services/storage.ts

```diff
@@ -41,7 +42,13 @@ class StorageService {
   public getStorageConnections = (): StorageConnection[] => {
     return this.storageConnections.map(s => {
       const temp = Object.assign({}, s);
-      temp.path = Path.resolve(s.path); // resolve path if path is relative, and change it to unix pattern
+      // if the last accessed path exist
+      if (fs.existsSync(s.path)) {
+        temp.path = Path.resolve(s.path); // resolve path if path is relative, and change it to unix pattern
+      } else {
+        this.createDefaultBotFolders();
```

Comment:
you are creating all folders inside a for loop. It's a O(n2) operation. 

my suggestion is call this method "ensureFolderExisit", and put this after this map.